lib/hotel.rb

```diff
@@ -0,0 +1,183 @@
+require_relative "room"
+require_relative "reservation"
+require_relative "date_range"
+require_relative "hotel_block"
+require "date"
+
+module HotelSystem
+  class Hotel
+    attr_reader :rooms, :blocks
+    attr_accessor :reservations
+
+    def initialize(number_of_rooms)
+      @rooms = []
+      add_rooms(number_of_rooms)
+      @reservations = []
+      @blocks = []
+    end
+
+    def add_rooms(number)
+      if number.class != Integer || number < 1
+        raise ArgumentError, "Please enter a number greater than 1."
+      end
```

Comment:
Again, avoid explicit type checking. If someone passes in the wrong type of thing, Ruby will produce an exception anyway.